plugin/pulpcore/plugin/stages/artifact_stages.py

```diff
@@ -205,9 +190,8 @@ def _downloaders_for_content(self, content):
                     declarative_artifact.url,
                     **validation_kwargs
                 )
-                downloaders_for_content.append(
-                    download_with_limit(self._download_semaphore, downloader)
-                )
+                downloaders_for_content.append(downloader.run())
```

Comment:
I don't think this works because you need to `await downloader.run()`. Calling downloader.run() returns the object of type `coroutine` which then needs to be scheduled "synchronously" via `await`. I didn't run the code, that is just what it seems like.

Reply:
@bmbouter, [`asyncio.gather`](https://github.com/pulp/pulp/blob/19b0c26bdd17e51fd468b332103bd6b9d96878ff/plugin/pulpcore/plugin/stages/artifact_stages.py#L162) is used for those.
I ran the code, it works for me. I still can miss something, so let me know :)

Reply:
Nope you are exactly correct! Thank you for pointing this out.